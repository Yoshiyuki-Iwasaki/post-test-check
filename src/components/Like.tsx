import React, { useState } from "react";
import styled from "styled-components";

const LikeButton = styled.span`
  background-color: rgb(231, 76, 60);
  color: white;
  padding: 0.5rem 0.5rem 0.5rem 1.5rem;
  border-radius: 0.4rem;
  cursor: pointer;
  position: absolute;
  bottom: 3px;
  left: 70px;
`;

const Like = () => {
  const [count, setCount] = useState<number>(0);
  const handleClick = () => {
    setCount(count + 1);
  };
  return (
    <LikeButton
      data-testid="counter"
      className="likeButton"
      onClick={handleClick}
    >
      {count}
    </LikeButton>
  );
};

export default Like;
