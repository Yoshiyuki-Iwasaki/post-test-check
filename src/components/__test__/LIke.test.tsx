import { cleanup, fireEvent, render, screen } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import Like from "../Like";

describe("いいね機能テスト", () => {

  afterEach(() => {
    cleanup();
  });

  // Snapshot テスト
  test("render", () => {
    const { asFragment } = render(<Like />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Click テスト
  test("Click テスト", () => {
    const component = render(<Like />);
    const likedButton = component.getByTestId("like");
    fireEvent.click(likedButton);
    screen.getByText("1");
    const unlikedButton = component.getByTestId("unlike");
    fireEvent.click(unlikedButton);
    screen.getByText("0");
  });
});
