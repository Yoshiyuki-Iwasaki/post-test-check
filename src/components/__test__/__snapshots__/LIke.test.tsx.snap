// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like テスト render 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL caPMGC"
    data-testid="like"
  >
    0
  </button>
</DocumentFragment>
`;
